### PR TITLE
[Reviewer: Alex] Allow non-registering PBXes to connect to Bono

### PR DESCRIPTION
**Pull request by rkd-msw**
Basically as discussed - the config and code is just parallel to the IBCF's trusted peers.

### CODE REVIEW

File: sprout/bono.cpp [hunk 1]
```diff
@@ -1378,6 +1401,17 @@ int proxy_process_access_routing(pjsip_rx_data *rdata,
...
           }
+          else
+          {
+            // Also check against the PBX peers.
+            TRC_DEBUG("Parsed destination as an IP address, and not a trusted peer, so check against PBX list");
+            if (is_pbx(dest))
+            {
+              TRC_DEBUG("Destination is a SIP PBX");
+              *trust = &TrustBoundary::OUTBOUND_EDGE_CLIENT;
+              pjsip_msg_find_remove_hdr(tdata->msg, PJSIP_H_AUTHORIZATION, NULL);
```
Comment (AlexHockey) [inline sprout/bono.cpp]: It seems odd to remove this here as I would expect the trust boundary to take care of it. I realize this is copied from above, but do you know why its required?

File: sprout/main.cpp [hunk 2]
```diff
@@ -357,6 +359,10 @@ static void usage(void)
        "                            in SIP (for example if a cluster of nodes is identified by the name\n"
        "                            'cluster.example.com', this value should be used instead of the hostnames\n"
        "                            or IP addresses of individual servers\n"
+       "     --non-registering-pbxes <comma-separated-list>\n"
+       "                            A comma separated list of domain names that are treated as\n"
```
Comment (AlexHockey) [inline sprout/main.cpp]: Are these domain names or IP addresses (the code below say IPs)?

File: sprout/pjutils.cpp [hunk 3]
```diff
@@ -447,6 +447,20 @@ void PJUtils::add_integrity_protected_indication(pjsip_tx_data* tdata, Integrity
...
+{
+  pjsip_proxy_authorization_hdr* auth_hdr = (pjsip_proxy_authorization_hdr*)
+                                      pjsip_msg_find_hdr(tdata->msg, PJSIP_H_PROXY_AUTHORIZATION, NULL);
+
+  if (auth_hdr == NULL)
+  {
+    auth_hdr = pjsip_proxy_authorization_hdr_create(tdata->pool);
+    auth_hdr->scheme = pj_str("Digest");
+    pjsip_msg_add_hdr(tdata->msg, (pjsip_hdr*)auth_hdr);
+  }
```
Comment (AlexHockey) [inline sprout/pjutils.cpp]: The spec says to add an empty response parameter - does this code do this?

File: sprout/ut/bono_test.cpp [hunk 4]
```diff
@@ -2020,6 +2051,35 @@ TEST_F(StatefulEdgeProxyAcceptRegisterTest, TestBonoEmergencyAcceptRegister)
...
+  // INVITE should be passed to Sprout despite the lack of a REGISTER
+  SCOPED_TRACE("INVITE (S)");
+  out = current_txdata()->msg;
+  ASSERT_NO_FATAL_FAILURE(ReqMatcher("INVITE").matches(out));
+  
+  // Goes to the configured upstream proxy ("upstreamnode", "10.6.6.8")
+  expect_target("TCP", "10.6.6.8", stack_data.pcscf_trusted_port, current_txdata());
+
+  // Check that a Proxy-Authorization header gets added.
+  EXPECT_EQ("Proxy-Authorization: Digest , nonce=\"\", response=\"\"",
```
Comment (AlexHockey) [inline sprout/ut/bono_test.cpp]: Checking the nonce parameter means the testcase is a little bit over-specified.

**Comment by AlexHockey** [comment]
Looks good. A few small things inline.

File: sprout/ut/bono_test.cpp (hunk 4 above)
Comment (rkd-msw) [inline sprout/ut/bono_test.cpp]: It is, but I can't see a good alternative.

Comment (AlexHockey) [inline sprout/ut/bono_test.cpp]: `MatchesRegex`?

**Pull request merged**